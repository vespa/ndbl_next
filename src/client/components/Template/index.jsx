import PropTypes from 'prop-types';
import React from 'react';
import 'normalize.css';
import Menu from '../Menu';
import Footer from '../Footer';

const Template = ({ children }) => {
  return (
    <React.Fragment>
      <Menu />
      {children}
      <Footer />
    </React.Fragment>
  );
};

Template.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Template;
