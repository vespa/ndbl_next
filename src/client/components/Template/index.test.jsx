import React from 'react';
import { shallow } from 'enzyme';
import Template from './index';

describe('[Component: FooterMenu]', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Template> test </Template>);
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
