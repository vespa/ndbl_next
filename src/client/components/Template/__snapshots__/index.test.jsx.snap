// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: FooterMenu] should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Template>
     test 
  </Template>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Menu />,
        " test ",
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      " test ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Menu />,
          " test ",
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " test ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
